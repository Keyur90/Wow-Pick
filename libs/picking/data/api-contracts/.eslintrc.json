{
  "extends": "../../../../.eslintrc.json",
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "parserOptions": {
        "project": ["libs/apps/picking/data/api-contracts/tsconfig.*?.json"]
      },
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
