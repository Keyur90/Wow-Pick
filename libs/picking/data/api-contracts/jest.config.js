module.exports = {
  displayName: 'picking-data-api-contracts',
  preset: '../../../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../../coverage/libs/picking/data/api-contracts',
};
