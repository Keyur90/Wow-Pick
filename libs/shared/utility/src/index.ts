export * from './barcode/GS1BarcodeParser';
export * from './barcode/parseScannedBarcode';
export * from './lib/bags';
export * from './lib/calculateShelfLifeExpiryDate';
export * from './lib/calculateTotalSuppliedQuantity';
export * from './lib/calculateTotalSuppliedQuantityForCollectible';
export * from './lib/calculateTrafficLightStatus';
export * from './lib/camelCase';
export * from './lib/canDispatchOrder';
export * from './lib/constants';
export * from './lib/dates';
export * from './lib/findIndexOfLastUnSuppliedItem';
export * from './lib/getEmptyTotes';
export * from './lib/hasAlreadySuppliedQuantity';
export * from './lib/hooks/useInputValidation';
export * from './lib/hooks/useSecondaryToleranceValidation';
export * from './lib/hooks/useTrafficLightStatus';
export * from './lib/isScannedBarcodeMatchWithArticleBarcode';
export * from './lib/isWeightedQuantityArticle';
export * from './lib/printLabels';
export * from './lib/quantityRange';
export * from './lib/returnToLegacyTrolleyMain';
export * from './lib/units';
export * from './lib/updateSuppliedQuantities';
export * from './lib/validateBarcode';
export * from './lib/validateExpiryDate';
export * from './lib/validateWeightTolerance';
export * from './types/barcode-data';
export * from './types/barcode-type';
export * from './types/BarcodeValidationResponse';
export * from './types/expiry-date-status';
export * from './types/trip-type';
