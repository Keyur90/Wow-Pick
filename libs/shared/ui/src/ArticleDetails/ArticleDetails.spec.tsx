import { ToteItem } from '@rf2/picking/data/api-contracts';
import { CollectOrder } from '@rf2/shared/utility';
import { render } from '@testing-library/react';
import React from 'react';
import { ArticleDetails } from './index';

const item: ToteItem = {
  id: '123',
  toteId: '111',
  itemIndex: 11,
  orderNo: '1234',
  articleId: 'AF1247',
  orderedQuantity: 7,
  stockOnHand: 235,
  article: { id: 'Af1247', brand: 'Coke', description: 'description', volumeSize: '500gm' },
  totalSuppliedQuantity: 5,
  suppliedDetails: [
    {
      type: 'primary',
      articleId: 'Af1247',
      scanDetails: [
        {
          barcode: '',
          suppliedPrice: 0,
          quantity: 5,
          weight: null,
          batchNo: '',
          expiryDate: '',
          dataBar: '',
        },
      ],
    },
  ],
  price: 10,
};

const order = {
  id: '1234',
  packagingType: 'ONLINE BAG',
  deliveryMethod: CollectOrder,
  locationCode: 5,
  deliveryDate: '2021-03-18T16:00:00',
  customerName: 'Suman',
};
const isRfRedesignPickToPictureEnabled = true;

describe('render Article Details', () => {
  it('should render ArticleDetails', async () => {
    const { container } = render(
      <ArticleDetails
        order={order}
        toteItem={item}
        isRfRedesignPickToPictureEnabled={isRfRedesignPickToPictureEnabled}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
