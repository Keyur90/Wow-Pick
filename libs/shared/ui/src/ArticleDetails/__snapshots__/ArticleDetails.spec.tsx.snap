// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Article Details should render ArticleDetails 1`] = `
<div>
  <div
    class="makeStyles-content-1"
  >
    <div
      class="makeStyles-description-2"
    >
      <div
        class="MuiContainer-root makeStyles-root-8 MuiContainer-maxWidthLg"
      >
        <div
          class="LinesEllipsis  "
        >
          Coke   500gm
          <wbr />
        </div>
      </div>
    </div>
    <div
      class="MuiContainer-root makeStyles-root-8 MuiContainer-maxWidthLg"
    >
      <div
        class="makeStyles-details-3"
      >
        <div
          class="makeStyles-articleId-4"
        >
          AF1247
        </div>
        <div
          class="makeStyles-amountDetails-5"
        >
          <span />
        </div>
        <div
          class="makeStyles-imageButtonWrapper-6"
        >
          <button
            aria-label="article"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-imageButton-7"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 19V5c0-1.1-.9-2-2-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2zM8.5 13.5l2.5 3.01L14.5 12l4.5 6H5l3.5-4.5z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
